src/api.h

```diff
@@ -108,9 +108,9 @@ namespace librealsense
     static void translate_exception(const char * name, std::string args, rs2_error ** error)
     {
         try { throw; }
-        catch (const librealsense_exception& e) { if (error) { rs2_free_error(*error); *error = rs2_create_error(e.what(), name, args.c_str(), e.get_exception_type() ); } }
-        catch (const std::exception& e)         { if (error) { rs2_free_error(*error); *error = rs2_create_error(e.what(), name, args.c_str(), RS2_EXCEPTION_TYPE_COUNT); } }
-        catch (...)                             { if (error) { rs2_free_error(*error); *error = rs2_create_error("unknown error", name, args.c_str(), RS2_EXCEPTION_TYPE_COUNT); } }
+        catch (const librealsense_exception& e) { if (error && !*error) *error = rs2_create_error(e.what(), name, args.c_str(), e.get_exception_type() ); }
```

Comment:
Personally, I'm not a big fan...
This opens up the possibility of uninitialized stack variables will be used before error would be detected and handled. 
But this is an issue with C style error checking in general, so I do see how this is convenient.

Reply:
```C++
rs2_error *e=nullptr;
rs2_get_frame_width(nullptr, &e);
rs2_get_Frame_height(nullptr,&e);
```
There are three things that could happen at the end of this sequence:

1. `e` has the second error and the first has been leaked (`origin/development`)
1. `e` has the second error and the first was freed (054cdec)
1. `e` has the first error and the second was never created (d843ecb)

I don't see an uninitialized stack variable in any of these three cases.  Can you explain where that would come from?

BTW, I'm happy to separate these additional patches into their own PR.

Reply:
@dorodnic, I don't think you meant the case of functions that don't have a way to check to success or failure besides checking `e`
```C++
rs2_error e;
rs2_set_option(options, option, 0, &e); // no return value to check for success
if (e)
   rs2_free_error(e);
```
since this would crash currently and likewise for ignoring the return value
```c++
rs2_error *e; // random garbage
rs2_get_frame_width(frame, &e);
if (e) rs2_free_error(e);
```
because this would currently crash on success as well.  This case would also currently crash on success,
```c++
rs2_error *e;
if (!rs2_get_frame_width(frame,&e) || e)
   rs2_free_error(e);
```
so maybe you meant meant functions that you can check for failure without checking `e` like this
```
rs2_error *e;
if (!rs2_get_frame_height(frame,&e) && e)
   rs2_free_error(e);
```
which would currently work in both the success and failure cases, but crash in the success case with both of my suggestions because `e` is uninitialized.   The good thing here is this has a chance of being noticed quickly.

FWIW, my searching this repo shows no uninitialized uses of `rs2_error`.

If you are otherwise happy with the changes here, I'd be happy to move these two to their own PR, drop them completely, or you can just merge the commit before them by hand and we can continue the discussion here.

Reply:
Maybe we could take the error handling into another PR? 
I'm not sure why it seem to conflict with CI, but the failure seem persistent, while "Fix compilation with TRACE_API" was passing. 
As for my earlier comment, if I understand the change correctly, you are allowing to make several calls in sequence and only check error once in the end. My concern is methods that are supposed to fill-in out parameters, but user forgets to check e before these are used.

Reply:
I dropped the change in error semantics from this PR.